pkg/controller/storagecluster/cephcluster_test.go

```diff
@@ -13,70 +13,90 @@ import (
+			}
+		}
+
+		reconciler := createFakeStorageClusterReconciler(t, c.cc)
+		err := reconciler.ensureCephCluster(mockStorageCluster, reconciler.reqLogger)
+		assert.NoError(t, err)
+		if c.condition == "" {
+			expected := newCephCluster(mockStorageCluster, "", 3, log)
+			actual := newCephCluster(mockStorageCluster, "", 3, log)
+			err = reconciler.client.Get(nil, mockCephClusterNamespacedName, actual)
+			assert.NoError(t, err)
+			assert.Equal(t, expected.ObjectMeta.Name, actual.ObjectMeta.Name)
+			assert.Equal(t, expected.ObjectMeta.Namespace, actual.ObjectMeta.Namespace)
+			assert.Equal(t, expected.Spec, actual.Spec)
+		} else if c.condition == "noCondition" {
+
+			assert.NotEmpty(t, reconciler.conditions)
+			assert.Len(t, reconciler.conditions, 3)
+
+			expectedConditions := map[conditionsv1.ConditionType]corev1.ConditionStatus{
+				conditionsv1.ConditionAvailable:   corev1.ConditionFalse,
+				conditionsv1.ConditionProgressing: corev1.ConditionTrue,
+				conditionsv1.ConditionUpgradeable: corev1.ConditionFalse,
+			}
+			for cType, status := range expectedConditions {
+				found := assertCondition(reconciler.conditions, cType, status)
+				assert.True(t, found, "expected status condition not found", cType, status)
+			}
+
+		} else {
+			assert.Empty(t, reconciler.conditions)
+		}
 
-func TestEnsureCephClusterNoConditions(t *testing.T) {
-	cc := newCephCluster(mockStorageCluster, "", 3, log)
-	cc.ObjectMeta.SelfLink = "/api/v1/namespaces/ceph/secrets/pvc-ceph-client-key" //for test purpose
-	reconciler := createFakeStorageClusterReconciler(t, cc)
-	err := reconciler.ensureCephCluster(mockStorageCluster, reconciler.reqLogger)
-	assert.NoError(t, err)
-	assert.NotEmpty(t, reconciler.conditions)
-	assert.Len(t, reconciler.conditions, 3)
-
-	expectedConditions := map[conditionsv1.ConditionType]corev1.ConditionStatus{
-		conditionsv1.ConditionAvailable:   corev1.ConditionFalse,
-		conditionsv1.ConditionProgressing: corev1.ConditionTrue,
-		conditionsv1.ConditionUpgradeable: corev1.ConditionFalse,
-	}
-	for cType, status := range expectedConditions {
-		found := assertCondition(reconciler.conditions, cType, status)
-		assert.True(t, found, "expected status condition not found", cType, status)
 	}
 }
 
-func TestEnsureCephClusterNegativeConditions(t *testing.T) {
-	cc := newCephCluster(mockStorageCluster, "", 3, log)
-	cc.ObjectMeta.SelfLink = "/api/v1/namespaces/ceph/secrets/pvc-ceph-client-key"
-	cc.Status.State = rookCephv1.ClusterStateCreated
-	reconciler := createFakeStorageClusterReconciler(t, cc)
-	err := reconciler.ensureCephCluster(mockStorageCluster, reconciler.reqLogger)
-	assert.NoError(t, err)
-	assert.Empty(t, reconciler.conditions)
-}
-
+<<<<<<< HEAD
 func TestStorageClusterCephClusterCreation(t *testing.T) {
 	// if both monPVCTemplate and monDataDirHostPath is provided via storageCluster
+=======
+func TestNewCephClusterMonData(t *testing.T) {
+>>>>>>> 81b22226... Changed function names
```

Comment:
You missed a conflict here. You want to accept the old version (`TestStorageClusterCephClusterCreation()`) and then apply the new version in a later commit. You can accomplish this during a `git rebase -i` by editing the commit to remove everything except line 95, then resolving any conflicts in later commits.

Reply:
Fixed the conflict. Let me know if any further changes are required. Thanks